### PR TITLE
Add Environment Variables documentation to the book

### PR DESCRIPTION
**Pull request by satk0**
It aims to resolve the rizin issue rizinorg/rizin#4445, but its **NOT** yet ready to merge !

While trying to solve it, I've met 2 problems:
- I don't know where to put rizin env vars doc (in a `16. Build Variables` or create a new chapter?) 
- `rz-test` is not documented, so nowhere to put env var doc (should I also document `rz-test`?)

### CODE REVIEW

**Comment by wargio** [comment]
Yes add a new section called `Run-Time Variables` after the `Build Variables` and probably we could rename `Build Variables` to `Compile-Time Variables`

**Comment by wargio** [comment]
@XVilka @kazarmy what do you think?

**Comment by wargio** [comment]
Regarding `rz-test`, it's not mapped in the book, because it is supposed to not be used by a user. its usage is available on the rizin repo itself: https://github.com/rizinorg/rizin/tree/dev/test#usage

**Comment by satk0** [comment]
Thanks for the answer. So I will just omit `rz-test` and focus on adding the new section and renaming `Build Variables` one.

**Comment by satk0** [comment]
Done, let me know how it looks

File: _quarto.yml [hunk 1]
```diff
@@ -38,7 +38,8 @@ book:
          - src/configuration/colors.md
          - src/configuration/evars.md
          - src/configuration/initial_scripts.md
-         - src/configuration/build_variables.md
+         - src/configuration/compile-time_variables.md
+         - src/configuration/run-time_variables.md
```
Comment (wargio) [inline _quarto.yml:42]: maybe use underscores instead of dashes for the files.

File: src/configuration/run-time_variables.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+## Run-Time Variables
+
+After successful compilation, run-time variables will be exposed to change the default behavior of rizin.
```
Comment (wargio) [inline src/configuration/run-time_variables.md:3]:
```suggestion
After successful compilation, run-time variables (also known as environment variables) will be exposed to change the default behavior of rizin.
```

File: src/configuration/run-time_variables.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ RZ_LOGSHOWSOURCES   should the log output contain src info (filename:lineno)
+ RZ_PIPE_IN          rzpipe cmd input (file descriptor)
+ RZ_PIPE_OUT         rzpipe cmd output (file descriptor)
+ RZ_MAGICPATH        /home/adam/.local/share/rizin/magic
+ RZ_NOPLUGINS        do not load rizin shared plugins
+ RZ_RCFILE           /home/adam/.rizinrc (user preferences, batch script)
+ RZ_DATAHOME         /home/adam/.local/share/rizin
+ RZ_VERSION          contains the current version of rizin
+ SFLIBPATH           SFLib syscall library path
+...
```
Comment (wargio) [inline src/configuration/run-time_variables.md:33]: if is not the full list, please add the complete one.

**Comment by satk0** [comment]
Done, that's the whole list of env vars that you can print with `-hh` command.

**Review by wargio — APPROVED** [review]

**Pull request merged**